src/java/org/apache/cassandra/index/sai/disk/hnsw/VectorPostingsWriter.java

```diff
@@ -24,12 +24,30 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.Function;
+
+import com.google.common.collect.BiMap;
 
 import org.apache.cassandra.io.util.SequentialWriter;
 import org.apache.cassandra.utils.Pair;
 
 public class VectorPostingsWriter<T>
 {
+
+    private final boolean useOrdinalsMapper;
+
+    private final Function<Integer, Integer> reverseOrdinalsMapper;
+
+    public VectorPostingsWriter() {
+        this.useOrdinalsMapper = false;
+        this.reverseOrdinalsMapper = x -> x;
```

Comment:
this is what i meant in the other comment -- can remove (almost) all the if statements since you have a mapper set up either way